compiler/rustc_trait_selection/src/solve/inspect/analyse.rs

```diff
@@ -343,8 +357,17 @@ impl<'a, 'tcx> InspectGoal<'a, 'tcx> {
             result,
             evaluation_kind: evaluation.kind,
             normalizes_to_term_hack,
+            source,
         }
     }
+
+    pub(crate) fn visit_with<V: ProofTreeVisitor<'tcx>>(&self, visitor: &mut V) -> V::Result {
```

Comment:
can you change https://doc.rust-lang.org/nightly/nightly-rustc/rustc_trait_selection/solve/inspect/struct.InspectCandidate.html#method.visit_nested_no_probe to use `visit_with` instead of handrolling their depth check? 🤔

Reply:
Oh yeah good idea